geo/src/algorithm/simplify_vw.rs

```diff
@@ -241,7 +241,7 @@ fn vwp_wrapper<T, const INITIAL_MIN: usize, const MIN_POINTS: usize>(
     epsilon: &T,
 ) -> Vec<Vec<Coord<T>>>
 where
-    T: CoordFloat + RTreeNum + HasKernel,
+    T: GeoFloat + RTreeNum,
```

Comment:
Without changing this change I get:
```
error[E0599]: no method named `intersects` found for struct `CachedEnvelope<geo_types::Line<T>>` in the current scope
```

This error was happening because the previous constraint, `CoordFloat + HasKernel` was enough to hit the `impl GeoNum for T` blanket impl. Since GeoFloat is basically `CoordFloat + HasKernel` it makes sense to just use GeoFloat to fix this and similar future problems if we add other traits.